src/UserManager.test.ts

```diff
@@ -161,7 +161,14 @@ describe("UserManager", () => {
     describe("signinRedirect", () => {
         it("should redirect the browser to the authorize url", async () => {
             // act
-            await subject.signinRedirect();
+            const spy = jest.fn();
+            subject.signinRedirect().finally(spy);
+
+            // signinRedirect is a promise that will never resolve (since we
+            // want it to hold until the page has redirected), so we use a
+            // timeout to wait for the majority of the code to have been run
+            // before proceeding with the rest of the test.
+            await new Promise(resolve => setTimeout(resolve, 200));
```

Comment:
Could you replace this with a promise waiting for the window to unload

```ts
await once(window, 'unload')
```

and update the mock implementation of the `unload` event to happen after a 200ms delay instead?

Reply:
Sorry, I probably should have suggested the import path too! 

```js
// https://nodejs.org/api/events.html#eventsonceemitter-name-options
import { once } from 'events';
```


It should be interoperable with browser `EventTarget` interfaces. If that doesn't work, what you have looks good 👍

Reply:
Ahh ok, didn't know about that! Much nicer looking than my version 👍🏾